### ISSUE TITLE
Title bar unresponsive after video mode change

### ISSUE THREAD

**Issue by SDLBugzilla** [libsdl-org/SDL-1.2#528]
# This bug report was migrated from our old Bugzilla tracker.

These attachments are available in the static archive:

* [Working but probably non-elegant patch (mouse_pressed_patch.diff, text/plain, 2009-11-04 04:52:39 +0000, 1839 bytes)](https://bugzilla.libsdl.org/attachment.cgi?id=443)

**Reported in version:** HG 1.2
**Reported for operating system, platform:** Windows (All), x86

# Comments on the original bug report:

On 2009-11-04 04:51:46 +0000, Gabriel Gambetta wrote:

> If you change the video mode while holding the mouse button down, and then click on the window, you can't move the mouse pointer over the title bar or the close window button.
> 
> It turns out WinMessage in SDL_Sysevents.c is using a static int mouse_pressed to keep track of whether it should call SetCapture() and ReleaseCapture(). Since it's static and initialized only once, it isn't cleared when the video mode changed, so there's a kind of one-off error and SetCapture() and ReleaseCapture() aren't being called when they should.
> 
> Here's a patch - I just made that int accessible from the outside and reset it to 0 in SDL_SetVideoMode, wrapped in #ifdef WIN32. Suggestions on how to make this more elegant are welcome.

On 2009-11-04 04:52:39 +0000, Gabriel Gambetta wrote:

> Created attachment 443
> Working but probably non-elegant patch

On 2009-11-04 16:04:59 +0000, Sam Lantinga wrote:

> Yes, working but definitely not elegant.
> 
> Thanks! :)

On 2011-12-30 01:25:52 +0000, Ryan C. Gordon wrote:

> 
> Bumping priority on a few bugs that I would like examined more closely before 1.2.15 is finalized. This is not a promise that a bug will be fixed. We may close it with WONTFIX or WORKSFORME or something, but I just want to make sure attention is paid.
> 
> --ryan.

On 2011-12-30 03:23:32 +0000, Sam Lantinga wrote:

> This is fixed, thanks!
> http://hg.libsdl.org/SDL/rev/760a3162e180
